clisops/ops/average.py

```diff
@@ -88,6 +89,81 @@ def average_over_dims(
     return op.process()
 
 
+class AverageShape(Operation):
+    def _resolve_params(self, **params):
+        shape = params.get("shape")
+        variable = params.get("variable", None)
+
+        self.params = {"shape": shape, "variable": variable}
+
+        if not shape:
+            raise InvalidParameterValue(
+                "At least one area for averaging must be provided"
+            )
+
+    def _get_file_namer(self):
+        extra = f"_avg-{'shape'}"
```

Comment:
Why the `{'shape'}`  ?

Reply:
it is the simple (temporary?) solution to naming the file, else we would need to either get name information from the polygon file itself, or ask the user to name their polygon. I can implement both if deemed necessary.

Reply:
But then it would just be `extra="_avg-shape"` ?

Reply:
Ah, correct yes I'll make the change.